### PR TITLE
Part of #17670: Replace Typography with Text component in smart-transaction-status.js

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation
This PR is a part of the issue #17670 Replace Typography with Text component ; also
it is a part of the issue #18714 Replace deprecated design system typography consts with enums in the file:

- **ui\pages\swaps\smart-transaction-status\smart-transaction-status.js**

### Before
Screenshot (71)

### After

Screenshot (72)

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Comment by brad-decker** [comment]
Rebase here too 🤪

**Comment by PrgrmrHarshShukla** [comment]
I tried to resolve the conflicts, but it dismissed the reviews of @DDDDDanica and @pedronfigueiredo 
Did not know that.

I made the changes as per the latest issues and developments.
@brad-decker ,
Please check.

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 1]
```diff
@@ -369,13 +375,16 @@ export default function SmartTransactionStatusPage() {
...
             variant={TextVariant.bodySm}
+            as="h6"
             boxProps={{ marginLeft: 2 }}
           >
             {`~${destinationValue && Number(destinationValue).toFixed(5)} `}
           </Text>
           <Text
             color={TextColor.textAlternative}
-            variant={TextVariant.bodySmBold}
+            variant={TextVariant.bodySm}
```
Comment (DDDDDanica) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:385]: You can still keep `variant={TextVariant.bodySmBold}` here :)

Comment (PrgrmrHarshShukla) [reply ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:385]: Yes, @DDDDDanica ,my bad.
Doing that now.

**Comment by PrgrmrHarshShukla** [comment]
@georgewrmarshall , @brad-decker , @DDDDDanica ,
Kindly look into this PR.

I am looking at the new animations 👀

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 2]
```diff
@@ -441,6 +451,7 @@ export default function SmartTransactionStatusPage() {
         {description && (
           <Text
             variant={TextVariant.bodySm}
+            as="h6"
             boxProps={{ ...(blockExplorerUrl && { margin: [1, 0, 0] }) }}
```
Comment (georgewrmarshall) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:455]: Need to update boxProps here to use `marginTop={1}`

**Review by georgewrmarshall — COMMENTED** [review]
Left a suggestion

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js (hunk 2 above)
Comment (PrgrmrHarshShukla) [reply ui/pages/swaps/smart-transaction-status/smart-transaction-status.js]: Hey @georgewrmarshall ,

Am I supposed to do this?
Screenshot (78)

Kindly look into this.

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 3]
```diff
@@ -369,13 +374,15 @@ export default function SmartTransactionStatusPage() {
           <Text
             color={TextColor.textAlternative}
             variant={TextVariant.bodySm}
+            as="h6"
             boxProps={{ marginLeft: 2 }}
```
Comment (georgewrmarshall) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:378]: boxProps needs to be updated

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 4]
```diff
@@ -369,13 +374,15 @@ export default function SmartTransactionStatusPage() {
...
           >
             {`~${destinationValue && Number(destinationValue).toFixed(5)} `}
           </Text>
           <Text
             color={TextColor.textAlternative}
             variant={TextVariant.bodySmBold}
+            as="h6"
             boxProps={{ marginLeft: 1 }}
```
Comment (georgewrmarshall) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:386]: boxProps needs to be updated

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 5]
```diff
@@ -403,14 +410,16 @@ export default function SmartTransactionStatusPage() {
             <Text
               color={TextColor.textAlternative}
               variant={TextVariant.bodySm}
+              as="h6"
               boxProps={{ marginLeft: 1 }}
```
Comment (georgewrmarshall) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:414]: boxProps needs to be updated

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @PrgrmrHarshShukla, looks like there are still some instances of `boxProps` in this file that need to be migrated

File: ui/pages/swaps/smart-transaction-status/smart-transaction-status.js [hunk 6]
```diff
@@ -441,7 +451,8 @@ export default function SmartTransactionStatusPage() {
         {description && (
           <Text
             variant={TextVariant.bodySm}
-            boxProps={{ ...(blockExplorerUrl && { margin: [1, 0, 0] }) }}
+            as="h6"
+            marginTop={1}
```
Comment (georgewrmarshall) [inline ui/pages/swaps/smart-transaction-status/smart-transaction-status.js:455]: Sorry I missed the conditional
```suggestion
            marginTop={blockExplorerUrl && 1}
```

**Comment by PrgrmrHarshShukla** [comment]
Hey @georgewrmarshall ,
It is done.
👍

**Review by georgewrmarshall — APPROVED** [review]
Great work! 💯

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**